kotest-framework/kotest-framework-multiplatform-plugin-gradle/src/test/kotlin/io/kotest/framework/multiplatform/gradle/KotestMultiplatformCompilerGradlePluginSpec.kt

```diff
@@ -1,60 +1,74 @@
 import java.io.InputStreamReader
 import java.nio.file.Files
 import java.nio.file.Path
 import java.nio.file.Paths
 
+
 // Why don't we use Gradle's TestKit here?
 // It embeds a particular version of Kotlin, which causes all kinds of pain.
 // See https://youtrack.jetbrains.com/issue/KT-24327 for one example.
 class KotestMultiplatformCompilerGradlePluginSpec : ShouldSpec({
    val kotestVersion = System.getProperty("kotestVersion")
 
-   setOf(
+   val kotlinVersions = setOf(
       "1.6.21",
-      "1.7.0"
-   ).forEach { kotlinVersion ->
-      context("when the project targets Kotlin version $kotlinVersion") {
-         val testProjectPath = Paths.get("src", "test", "resources", "test-project").toAbsolutePath()
-         val testReportsDirectory = testProjectPath.resolve("build").resolve("test-results")
+      "1.7.0",
+   )
+
+   beforeAny {
+      // Build the test project ahead of time so that it doesn't try to build the Kotest libraries during
+      // the tests. (if it tries to build them during the tests this can lead to conflicts).
+      kotlinVersions.forEach { kotlinVersion ->
+         GradleInvocation(
+            "-PkotlinVersion=$kotlinVersion",
+            "-PkotestVersion=$kotestVersion",
+            "-PuseNewNativeMemoryModel=false",
+            "build",
+            "-x", "check",
+         ).run()
+      }
+   }
 
-         beforeEach {
-            if (Files.exists(testReportsDirectory)) {
-               if (!testReportsDirectory.toFile().deleteRecursively()) {
-                  throw RuntimeException("Could not delete test report directory $testReportsDirectory")
-               }
-            }
+   beforeEach {
+      if (Files.exists(testReportsDirectory)) {
+         if (!testReportsDirectory.toFile().deleteRecursively()) {
+            throw RuntimeException("Could not delete test report directory $testReportsDirectory")
          }
+      }
+   }
+
+   kotlinVersions.forEach { kotlinVersion ->
+      context("when the project targets Kotlin version $kotlinVersion") {
 
          fun shouldHavePassingTestResultsFor(taskName: String) {
             val testReportFile = testReportsDirectory.resolve(taskName).resolve("TEST-TestSpec.xml")
             testReportFile.toFile().shouldBeAFile()
 
-            val testReportContents = Files.readAllBytes(testReportFile).decodeToString()
+            val testReportContents: String = Files.readAllBytes(testReportFile).decodeToString()
 
-            testReportContents shouldStartWith """
-               <?xml version="1.0" encoding="UTF-8"?>
-               <testsuite name="TestSpec" tests="2" skipped="0" failures="0" errors="0"
-            """.trimIndent()
+            testReportContents.shouldContainInOrder(
+               """  <?xml version="1.0" encoding="UTF-8"?>                                    """.trim(),
+               """  <testsuite name="TestSpec" tests="2" skipped="0" failures="0" errors="0"  """.trim(),
+            )
```

Comment:
Is there a reason for this change?

Reply:
The previous assertion failed on Windows because of line endings differences